backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/InitVdsOnUpCommand.java

```diff
@@ -201,24 +206,42 @@ private void logHeHostInForeignClusterAudit(Cluster cluster) {
 
     private boolean initVirtResources() {
         resourceManager.clearLastStatusEventStampsFromVds(getVdsId());
-        if (initializeStorage()) {
+        if (checkFipsMode() && initializeStorage()) {
```

Comment:
wouldn't it be better to have it `initializeStorage() && checkFipsMode()`? to make sure we first have connection to the SD? or even within the `if` statement, like it was before?

Reply:
Why is FIPS mode related only to virt and not gluster? Shouldn't it be relevant to both?
I'd probably make FIPS as a standalone check in executeCommand() before trusted services check:

```
...
    initSucceeded = checkFipsMode();
    if (initSucceeded && cluster.supportsTrustedService()) {
        initSucceeded = initTrustedService();
    }
...
```

Reply:
> wouldn't it be better to have it `initializeStorage() && checkFipsMode()`? to make sure we first have connection to the SD? or even within the `if` statement, like it was before?

one downside of github is that it's easy to miss the text that is associated with each commit. I touched on that [in the text of the last commit](https://github.com/oVirt/ovirt-engine/pull/255/commits/8466bd83b443fd05ac2b099d666948dcdc8456bd) - checking the FIPS mode is significantly faster than connecting the storage domain, if we are anyway going to set the host as non-operational we can avoid/postpone the latter

Reply:
> Why is FIPS mode related only to virt and not gluster? Shouldn't it be relevant to both? I'd probably make FIPS as a standalone check in executeCommand() before trusted services check:
> 
> ```
> ...
>     initSucceeded = checkFipsMode();
>     if (initSucceeded && cluster.supportsTrustedService()) {
>         initSucceeded = initTrustedService();
>     }
> ...
> ```

yeah, seems gluster is compatible with FIPS now so it sounds like a good idea